### PR TITLE
Feature/its all about fonts

### PR DESCRIPTION
**Pull request by Nats-ji**
- [x] Icons for overlay windows and settings widget
- [x] Vertical align the icon font
- [x] Emoji font 🤣 because why not (Windows 10 and above)
- [x] Monospace font for the console
- [x] Expose monospace font to mods `ImGui.GetMonoFont()`, `ImGui.PushFont()` #768
- [x] Option for custom monospace font
- [x] Reload font during runtime
- [x] GUI for changing font settings #488
- [x] Enumerate system font
- [x] Remove the glyph range option. ~Instead, load every glyph in the supplied font.~
- [x] Merge the default NotoSans fonts for better multilingual support.
- [x] Logs
- ~Font preview~ (maybe later)
- [x] Dynamic glyph range, instead of loading all glyphs

Dynamic glyph ranges todo list:
- [x] console logs
- [x] console inputtext 
- [x] lua binding inputtext wrapper
- [x] expose `AddTextGlyphs()` to lua
- [x] pre-cache ".lua", ".txt", ".json", ".yml", ".yaml", ".toml", ".ini" files in mod folders
- [x] pre-cache all path strings in the mod folders
- [x] dynamically load icon glyphs from mods
- [x] add files opened by mods (`io.open`, `io.lines`) to glyph ranges builder

Use monospace font from lua:
```lua
ImGui.PushFont(ImGui.GetMonoFont())
ImGui.Text("This is how the monospaced font looks like")
ImGui.PopFont()
```
Display multilingual string:
image

New settings panel:
image

### CODE REVIEW

**Comment by maximegmd** [comment]
Do you consider this complete?

**Comment by Nats-ji** [comment]
> Do you consider this complete?

yes

File: src/Fonts.cpp [hunk 1]
```diff
@@ -0,0 +1,505 @@
...
+    const bool useCustomMainFont = !customMainFontPath.empty();
+    const bool useCustomMonosFont = !customMonosFontPath.empty();
+
+    const auto defaultMainFontPath = GetAbsolutePath(m_defaultMainFont, m_paths.Fonts(), false);
+    const auto defaultMonoFontPath = GetAbsolutePath(m_defaultMonoFont, m_paths.Fonts(), false);
+
+    // Set main font path to default if customMainFontPath is empty or doesnt exist.
+    const auto mainFontPath = useCustomMainFont ? customMainFontPath : defaultMainFontPath;
+    if (mainFontPath.empty())
+        Log::Error("Can't find default main font at path: {}, will use built-in font.", GetAbsolutePath(m_defaultMainFont, m_paths.Fonts(), true).string());
```
Comment (maximegmd) [inline src/Fonts.cpp:125]: Probably best to trigger warnings if it's not breaking anything and has a fallback

**Review by maximegmd — CHANGES_REQUESTED** [review]
Apart from the log level, looks great !

File: src/Fonts.cpp (hunk 1 above)
Comment (Nats-ji) [reply src/Fonts.cpp]: Done

**Review by maximegmd — APPROVED** [review]

**Pull request merged**